src/main/java/net/sf/jabref/shared/DBMSType.java

```diff
@@ -76,13 +75,26 @@ public int getDefaultPort() {
     }
 
     public static Optional<DBMSType> fromString(String typeName) {
-        for (DBMSType dbmsType : EnumSet.allOf(DBMSType.class)) {
-            if (typeName.equals(dbmsType.toString())) {
-                return Optional.ofNullable(dbmsType);
-            }
+        try {
+            return Optional.of(Enum.valueOf(DBMSType.class, typeName.toUpperCase()));
```

Comment:
Good! But I recommend adding Locale.ENGLISH in the toUpperCase method.
Here is a nice explanation why it is sometimes necessary (the infamous Turkish Locale bug)
http://mattryall.net/blog/2009/02/the-infamous-turkish-locale-bug